src/Zipper.elm

```diff
@@ -20,41 +24,84 @@ type alias Zipper a =
     }
 
 
-makeZipper : List a -> a -> List a -> Maybe (Zipper a)
-makeZipper =
-    Debug.todo "make Zipper"
+makeZipper : List a -> a -> List a -> Zipper a
+makeZipper previous current next =
+    { previous = previous, current = current, next = next }
+
+
+moveBackwards : Zipper a -> Maybe (Zipper a)
+moveBackwards zipper =
+    case init zipper.previous of
+        Nothing ->
+            Nothing
+
+        Just choppedPrevious ->
+            Just <|
+                { previous = choppedPrevious
+                , current = withDefault zipper.current <| last zipper.previous
+                , next = zipper.current :: zipper.next
+                }
 
 
 moveForward : Zipper a -> Maybe (Zipper a)
-moveForward =
-    Debug.todo "make moveForward"
+moveForward zipper =
+    map flipZipp << moveForward << flipZipp <| zipper
```

Comment:
Or with a point-free style:
```elm
moveForward : Zipper a -> Maybe (Zipper a)
moveForward =
    map flipZipp << moveForward << flipZip
```

Reply:
Yeah, but then it catches that it's a cyclic definition

Reply:
Nvm